libraries/AC_Autorotation/AC_Autorotation.cpp

```diff
@@ -696,19 +698,28 @@ void AC_Autorotation::touchdown_controller(void)
     // Update forward speed for logging
     _speed_forward = calc_speed_forward(); // (cm/s)
 
-    _collective_out =  constrain_value((_p_coll_tch.get_p(_desired_sink_rate - _current_sink_rate))*0.01f + _ff_term_hs, 0.0f, 1.0f);
-    col_trim_lpf.set_cutoff_frequency(_col_cutoff_freq);
-    _ff_term_hs = col_trim_lpf.apply(_collective_out, _dt);
-    if(_current_sink_rate < 10.0 && _radar_alt<=_ground_clearance && !_touchdown_complete){
+    // Check to see if we think the aircraft is on the ground
+    if(_current_sink_rate < 10.0 && _radar_alt<=_ground_clearance && _time_on_ground == 0){
         _time_on_ground = AP_HAL::millis();
-        _touchdown_complete = true;
     }
-    float now = AP_HAL::millis();
-    if((now - _time_on_ground) > MIN_TIME_ON_GROUND && _touchdown_complete){
-       //on ground, collective can be bottomed now
-       _collective_out = 0;
+
+    // Use a timer to get confidence that the aircraft is on the ground.
+    // Note: The landing detector uses the zero thust collective as an indicator for being on the ground. The touch down controller will have
+    // driven the collective high to cushion the landing so the landing detector will not trip until we drive the collective back to zero thrust and below.
+    if ((_time_on_ground > 0) && ((AP_HAL::millis() - _time_on_ground) > MIN_TIME_ON_GROUND)) {
+       // On ground, smoothly lower collective to just bellow zero thrust, to make sure we trip the landing detector
+       float desired_col = _motors_heli->get_coll_mid() * 0.95;
```

Comment:
This needs changed to use H_COL_LAND_MIN instead of zero thrust collective, since the landing detector looks for "motors->get_below_land_min_coll()"